File: BusinessCardHolder/Java/Personal.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package businesscardholder;

/**
 *
 * @author Taehoo
 */
public class Personal {
    
    public Personal(){
        this.name = "";
        this.telephoneNumber = "";
        this.emailAddress = "";
        this.position = "";
    }
    
    public Personal(String name, String telephoneNumber, String emailAddress, 
            String position){
        this.name = name;
        this.telephoneNumber = telephoneNumber;
        this.emailAddress = emailAddress;
        this.position = position;
    }
    
     public boolean IsEqual(Personal other){
        boolean ret = false;
        
        if(this.name.compareTo(other.name) ==0 && this.telephoneNumber.compareTo(other.telephoneNumber) ==0 &&
                this.emailAddress.compareTo(other.emailAddress) == 0 && this.position.compareTo(other.position) == 0){
            ret = true;
        } 
        
        return ret;
    } 
    
    public boolean IsNotEqual(Personal other){
        boolean ret = false;
        
       if(this.name.compareTo(other.name) !=0 || this.telephoneNumber.compareTo(other.telephoneNumber) !=0 ||
                this.emailAddress.compareTo(other.emailAddress) !=0 || this.position.compareTo(other.position) !=0){
            ret = true;
        } 
       
       return ret;
    }
    
    public String GetName(){
        return this.name;
    }
    public String GetTelephoneNumber(){
        return this.telephoneNumber;
    }
    public String GetEmailAddress(){
        return this.emailAddress;
    }
    public String GetPosition(){
        return this.position;
    }
   
    private String name;
    private String telephoneNumber;
    private String emailAddress;
    private String position;
}

File: BusinessCardHolder/Java/Personal.cpp
#include "Personal.h"

Personal::Personal()
	: name(""), telephoneNumber(""), emailAddress(""), position("") {
}
Personal::Personal(string name, string telephoneNumber, string emailAddress, string position)
	: name(name), telephoneNumber(telephoneNumber), emailAddress(emailAddress), position(position) {
}
Personal::Personal(const Personal& source)
	: name(source.name), telephoneNumber(source.telephoneNumber), emailAddress(source.emailAddress), position(source.position) {
}
Personal::~Personal() {
}
bool Personal::IsEqual(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) == 0 &&
		this->telephoneNumber.compare(other.telephoneNumber) == 0 &&
		this->emailAddress.compare(other.emailAddress) == 0 &&
		this->position.compare(other.position) == 0) {
		ret = true;
	}

	return ret;
}
bool Personal::IsNotEqual(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) != 0 ||
		this->telephoneNumber.compare(other.telephoneNumber) != 0 ||
		this->emailAddress.compare(other.emailAddress) != 0 ||
		this->position.compare(other.position) != 0) {
		ret = true;
	}

	return ret;
}
bool Personal::operator ==(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) == 0 &&
		this->telephoneNumber.compare(other.telephoneNumber) == 0 &&
		this->emailAddress.compare(other.emailAddress) == 0 &&
		this->position.compare(other.position) == 0) {
		ret = true;
	}

	return ret;
}
bool Personal::operator !=(const Personal& other) {
	bool ret = false;
	if (this->name.compare(other.name) != 0 ||
		this->telephoneNumber.compare(other.telephoneNumber) != 0 ||
		this->emailAddress.compare(other.emailAddress) != 0 ||
		this->position.compare(other.position) != 0) {
		ret = true;
	}

	return ret;
}
Personal& Personal::operator =(const Personal& source) {
	this->name = source.name;
	this->telephoneNumber = source.telephoneNumber;
	this->emailAddress = source.emailAddress;
	this->position = source.position;

	return *this;
}